src/prefect/orion/api/work_queues.py

```diff
@@ -126,6 +131,17 @@ async def read_work_queue_runs(
             limit=limit,
         )
 
+        # The Prefect UI often calls this route to see which runs are enqueued.
+        # We do not want to record this as an actual poll event.
+        if not x_prefect_ui:
+            await models.work_queues.update_work_queue(
+                session=session,
+                work_queue_id=work_queue_id,
+                work_queue=schemas.actions.WorkQueueUpdate(
+                    last_polled=pendulum.now("UTC")
+                ),
+            )
+
```

Comment:
It feels like both this and the `record_agent_poll` below should be done in the background.  It would mean opening a new session for those updates, so there's a tradeoff there in terms of latency versus connection capacity.